src/ngraph/runtime/reference/allreduce.hpp

```diff
@@ -16,10 +16,12 @@
 
 #pragma once
 
-#ifdef NGRAPH_DISTRIBUTED
-
+#ifdef NGRAPH_DISTRIBUTED_ENABLE
+#ifdef NGRAPH_DISTRIBUTED_MLSL_ENABLE
```

Comment:
Instead of using MPI/MLSL both in this file - create two files - one for MPI and one for MLSL. Then include them based on the specific transport.

Reply:
I'll leave it as is for now (as it unifies the design on this PR), address this part of refactoring for this file and others in the consequent PR.